src/slre.c

```diff
@@ -1184,7 +1184,7 @@ static void re_newthread(struct slre_thread *t, struct slre_instruction *pc,
   if (!(thr = 0)) continue
 
 static unsigned char re_match(struct slre_instruction *pc, const char *start,
-                              const char *bol, unsigned char flags,
+                              const char *bol, unsigned flags,
```

Comment:
Throughout the rest of our code base we explicitly use `unsigned int`. AFAIK `unsigned` and `unsigned int` are synonyms.